pkg/operator/controller/controller_service_monitor.go

```diff
@@ -8,23 +8,36 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 
+	operatorclient "github.com/openshift/cluster-ingress-operator/pkg/operator/client"
+
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func (r *reconciler) ensureServiceMonitor(ci *operatorv1.IngressController, svc *corev1.Service, deploymentRef metav1.OwnerReference) (*unstructured.Unstructured, error) {
 	desired := desiredServiceMonitor(ci, svc, deploymentRef)
 
-	current, err := r.currentServiceMonitor(ci)
+	// ServiceMonitor CRD may not be created by the time r.Client is created
+	// and since r.Client will not refresh scheme/mapper fields after creation,
+	// ServiceMonitor GVK may not be recognized.
+	// So we use a new kube client everytime to get or create ServiceMonitor resource.
+	kubeClient, err := operatorclient.NewClient(r.KubeConfig)
```

Comment:
from my own local testing:

```
2019-03-15T11:17:10.143-0400    INFO    operator        log/log.go:26   started zapr logger
2019-03-15T11:17:10.154-0400    INFO    operator.entrypoint     cluster-ingress-operator/main.go:45     config.GetConfig()      {"duration": "1.61447ms"}                                                      
2019-03-15T11:17:12.155-0400    INFO    operator.entrypoint     cluster-ingress-operator/main.go:52     operator.Client()       {"duration": "2.000241625s"}
```

My suspicion is that we're introducing a 2s floor time to every call to ensureServiceMonitor

Reply:
Could we add something like `if errors.IsNoMatchError(err) { // recreate r.Client and retry` to the error check in `currentServiceMonitor`?